src/Api/Monetization/Entity/Property/StartDatePropertyAwareTrait.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apigee\Edge\Api\Monetization\Entity\Property;
+
+/**
+ * Trait StartDatePropertyAwareTrait.
+ *
+ * @see \Apigee\Edge\Api\Monetization\Entity\Property\StartDatePropertyInterface
+ */
+trait StartDatePropertyAwareTrait
+{
+    /**
+     * @var \DateTimeImmutable
+     */
+    protected $startDate;
+
+    /**
+     * @inheritdoc
+     */
+    public function getStartDate(): \DateTimeImmutable
```

Comment:
One issue I've had is when `\Drupal\apigee_edge\Entity\FieldableEdgeEntityBaseTrait::getOriginalFieldData` gets called on a new entity, there won't be any field values set yet.  This causes a `TypeError`. `::set` calls `::get` which calls `::getOriginalFieldData`.

Reply:
@Jaesin This is a Drupal 8 integration related issue probably, so if you need help about it please report it in the Drupal 8 module repo or in a private channel. We also had our fight with Drupal 8's typed data system when we worked on the Management API integration.